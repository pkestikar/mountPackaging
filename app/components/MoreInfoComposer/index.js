import React from 'react';
import Button from "Components/Button";

const MoreInfoComposer = (props) => {
    return (
        <div className="mp--more-info-composer">
            <span className={`${props.normalFontSize ? 'normal-font' : ''}`}>{props.title}</span>
            <Button palette={'secondary'} onClick={() => props.handleClick()}>{props.ctaText}</Button>
        </div>
    )
};

MoreInfoComposer.defaultProps = {
    title: '',
    ctaText: '',
    handleClick: () => {},
    normalFontSize: false
};

export default MoreInfoComposer;
