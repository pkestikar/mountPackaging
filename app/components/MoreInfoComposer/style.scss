.mp--more-info-composer {
  height: 350px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  &.normal-font {
    span {
      font-size: 1.2rem;
    }
  }
  &.border-top {
    border-top: 1px solid var(--light-periwinkle);
  }
  span {
    font-size: 1.8rem;
    margin-bottom: 2rem;
    text-align: center;
  }
  button {
    margin-top: 15px;
  }
  @media only screen and (max-width: $medium) {
    height: 200px;
    span {
      font-size: 1.4rem;
      margin-bottom: 0.9rem;
    }
  }
}
