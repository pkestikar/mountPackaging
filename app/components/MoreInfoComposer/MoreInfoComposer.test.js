import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import MoreInfoComposer from "Components/MoreInfoComposer/index";

let handleClick = jest.fn();

describe('MoreInfoComposer component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<MoreInfoComposer />);
    });

    it('renders correctly', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders a blank title if no title is provided in the props', () => {
        expect(wrapper.find('span').text()).toEqual('');
    });

    it('renders the cta with no text value if not provided', () => {
        expect(wrapper.find('Button').text()).toEqual('');
    });

    it('executes the function provided for the button', () => {
        wrapper.setProps({ handleClick: handleClick });
        wrapper.find('Button').simulate('click');
        expect(handleClick).toBeCalled();
    });

    it('renders the provided title and the cta text', () => {
        let title = 'This is the text title';
        let cta = 'Click here';

        wrapper.setProps({ title: title, ctaText: cta });
        expect(wrapper.find('span').text()).toEqual(title);
        expect(wrapper.find('Button').text()).toEqual(cta);
    });
});
