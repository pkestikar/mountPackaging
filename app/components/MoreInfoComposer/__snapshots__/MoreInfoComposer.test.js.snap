// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreInfoComposer component renders correctly 1`] = `
<MoreInfoComposer
  borderTop={false}
  ctaText=""
  handleClick={[Function]}
  normalFontSize={false}
  title=""
>
  <div
    className="mp--more-info-composer"
  >
    <span />
    <Button
      disabled={false}
      loading={false}
      onClick={[Function]}
      palette="secondary"
      size="medium"
    >
      <button
        className="mp--button medium mp--button-secondary"
        onClick={[Function]}
      />
    </Button>
  </div>
</MoreInfoComposer>
`;
